src/data/serializer/XYZ.js

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2015 The Open Source Geospatial Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ /**
+  * A serializer for layer that hava a `ol.source.XYZ` source.
+  */
+Ext.define('GeoExt.data.serializer.XYZ', {
+    extend: 'GeoExt.data.serializer.Base',
+    inheritableStatics: {
+        /**
+         * @inheritdoc
+         */
+        sourceCls: ol.source.XYZ,
+
+        /**
+         * @inheritdoc
+         */
+        serialize: function(layer, source) {
+            this.validateSource(source);
+            var tileGrid = source.getTileGrid();
+            var serialized = {
+                "baseURL": source.getUrls()[0],
```

Comment:
`getUrl()` may be [null](http://openlayers.org/en/master/apidoc/ol.source.XYZ.html#getUrls), so I'd add a check for this. It may be we do not have these cheks for the other serialzers, but why not start here.